src/index.ts

```diff
@@ -22,18 +27,8 @@ const csvDownload = ({
     return;
   }
 
-  const headerKeys = Object.keys(data[0]);
-
-  const columnNames = headers ?? headerKeys;
-  const csv = data.map((row) =>
-    headerKeys
-      .map((fieldName) =>
-        JSON.stringify(row[fieldName] === 0 ? 0 : row[fieldName] ?? "")
-      )
-      .join(delimiter)
-  );
-  csv.unshift(columnNames.join(delimiter));
-  const csvAsString = csv.join("\r\n");
```

Comment:
I moved this to a separate file so that it could be tested.

Reply:
Good call.